File: SkaiLab.Invoice.Models/Product.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace SkaiLab.Invoice.Models
{
    public class Product
    {
        public long Id { get; set; }
        public string OrganisationId { get; set; }
        public string Code { get; set; }
        public string Name { get; set; }
        public string ImageUrl { get; set; }
        public bool TrackInventory { get; set; }
        public ProductSalePurchaseDetail ProductPurchaseInformation { get; set; }
        public ProductSalePurchaseDetail ProductSaleInformation { get; set; }
        public List<InventoryHistory> InventoryHistories { get; set; }
        public long? LocationId { get; set; }
        public Location Location { get; set; }
        public int QtyBalance { get; set; }

    }
    public class ProductSalePurchaseDetail
    {
        public decimal Price { get; set; }
        public long? TaxId { get; set; }
        public Tax Tax { get; set; }
        public string Description { get; set; }
        public string Title { get; set; }
    }
}

File: SkaiLab.Invoice/ClientApp/src/app/models/product.ts
import { InventoryHistory } from "./inventoryHistory";
import { Tax } from "./tax";
import {Location} from './location';

export class Product{
    id:number;
    organisationId:string;
    code:string;
    name:string;
    imageUrl:string;
    trackInventory:boolean;
    productPurchaseInformation:ProductSalePurchaseDetail;
    productSaleInformation:ProductSalePurchaseDetail;
    inventoryHistories:InventoryHistory[];
    locationId:number;
    location:Location;
    qtyBalance:number;
    constructor(){
        this.id=0;
        this.organisationId="";
        this.code="";
        this.imageUrl="";
        this.trackInventory=false;
        this.productPurchaseInformation=new ProductSalePurchaseDetail();
        this.productSaleInformation=new ProductSalePurchaseDetail();
        this.inventoryHistories=[];
        this.locationId=null;
        this.location=new Location();
        this.qtyBalance=0;
    }
}
export class ProductSalePurchaseDetail{
    price:number;
    taxId?:number;
    tax:Tax;
    description:string;
    title:string;
    constructor(){
        this.price=0;
        this.taxId=null;
        this.tax=null;
        this.description="";
        this.title="";
    }
}